keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -45,5 +55,51 @@ class DeliveryConfigController(
   fun diff(@RequestBody deliveryConfig: SubmittedDeliveryConfig): List<EnvironmentDiff> =
     adHocDiffer.calculate(deliveryConfig)
 
+  @GetMapping(
+    path = ["/{name}/environment/{environment}/constraint-states"],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  fun getConstraintState(
+    @PathVariable("name") name: String,
+    @PathVariable("environment") environment: String,
+    @RequestParam("limit") limit: Int?
+  ): List<ConstraintState> =
+    deliveryConfigRepository.constraintStateFor(name, environment, limit ?: DEFAULT_MAX_EVENTS)
+
+  @PostMapping(
+    path = ["/{name}/environment/{environment}/constraint-state/{type}"],
+    consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  // TODO: Keel needs fiat integration and should use kork's AuthenticatedRequest class.
+  //  In this case, we should ensure the user has an appropriate role based on a service account
+  //  provided to keel. Service account may need to move to the environment or delivery config level.
+  fun updateConstraintStatus(
+    @RequestHeader("X-SPINNAKER-USER") user: String,
+    @PathVariable("name") name: String,
+    @PathVariable("environment") environment: String,
+    @PathVariable("type") type: String,
+    @RequestParam("artifactVersion") artifactVersion: String,
+    @RequestParam("status") status: String,
+    @RequestParam("comment") comment: String?
```

Comment:
@erikmunson I'm not sure if `/delivery-configs/{name}/environment/{environment}/constraint-state/{type}` plus the required request params makes for the best interface, let me know if you have any feedback on the new routes.

Reply:
I thought of suggesting to have a separate controller for the constraints I guess it doesn't make much sense since it relies on the delivery config name. Maybe just `constraint` instead of `constraint-state`?

Reply:
+1 to changing to route to `constraint`, will do. If we end up with an `EnvironmentController`, I think constraints will belong there, fwiw.